File: 167.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> hash = new HashMap<>(); 
        
        for (int i = 0 ; i < nums.length; i++){
            int val = hash.getOrDefault(target - nums[i], 0); 
            if (val > 0){
                return new int[]{val, i+1}; 
            } else {
                val = hash.getOrDefault(nums[i], 0); 
                if (val == 0) hash.put(nums[i], i+1); 
            }
        }
        
        return new int[]{}; 
    }
}

File: 167.cs
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/description/ 

public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int rp = 0, lp = numbers.Length-1;
        while(lp > rp) {
            if (numbers[lp] + numbers[rp] == target) 
                return new int[]{rp+1, lp+1}; 
            else if (numbers[lp] + numbers[rp] > target)
                lp -= 1; 
            else 
                rp += 1; 
        } 

        return new int[]{rp+1, lp+1}; 
    }
}
